graylog2-web-interface/src/views/components/messagelist/MessageFields/MessageFieldsEditModal.tsx

```diff
@@ -0,0 +1,119 @@
+import StreamLink from 'components/streams/StreamLink';
+import { useStore } from 'stores/connect';
+import { StreamsStore } from 'views/stores/StreamsStore';
+import useFormattedFields from 'views/components/messagelist/MessageFields/hooks/useFormattedFields';
+import MessageFavoriteFieldsContext from 'views/components/contexts/MessageFavoriteFieldsContext';
+
+const FieldsContainer = styled.div(
+  ({ theme }) => css`
+    overflow: visible;
+    padding: ${theme.spacings.xs} 0;
+  `,
+);
+
+const ModalContentContainer = styled.div`
+  max-height: 500px;
+  overflow-y: auto;
+  font-size: ${({ theme }) => theme.fonts.size.small};
+`;
+
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+  align-items: end;
+`;
+
+const StyledButtonGroup = styled(ButtonGroup)`
+  gap: ${({ theme }) => theme.spacings.xs};
+`;
+
+const MessageFieldsEditModal = ({ toggleEditMode }) => {
+  const { formattedFavorites, formattedRest } = useFormattedFields();
+  const { message, saveFavoriteField, resetFavoriteField, cancelEdit } = useContext(MessageFavoriteFieldsContext);
+  const messageStreams = useStore(StreamsStore, ({ streams }) =>
+    streams.filter((stream) => message.fields.streams.includes(stream.id)),
+  );
+
+  const _saveFavoriteField = useCallback(() => {
+    saveFavoriteField();
+    toggleEditMode();
+  }, [saveFavoriteField, toggleEditMode]);
+
+  const _resetFavoriteField = useCallback(() => {
+    resetFavoriteField();
+    toggleEditMode();
+  }, [resetFavoriteField, toggleEditMode]);
+
+  const cancelModal = useCallback(() => {
+    cancelEdit();
+    toggleEditMode();
+  }, [cancelEdit, toggleEditMode]);
+
+  return (
+    <Modal onHide={cancelModal} show>
+      <Modal.Header>
+        <Modal.Title>Favorite fields configuration</Modal.Title>
+      </Modal.Header>
+      <Modal.Body>
+        <ModalContentContainer>
+          <FieldsContainer>
+            <Alert bsStyle="info">
+              Favorite fields will be applied to the streams{' '}
+              {messageStreams.map((stream, index) => {
+                const isLast = index === messageStreams.length - 1;
+
+                return (
+                  <React.Fragment key={stream.id}>
+                    <StreamLink key={stream.id} stream={stream} />
+                    {!isLast && ', '}
+                  </React.Fragment>
+                );
+              })}{' '}
+              in which this message is routed.
+            </Alert>
+            <MessageFieldsEditModeList fields={formattedFavorites} message={message} isFavorite />
+            <MessageFieldsEditModeList fields={formattedRest} message={message} isFavorite={false} />
+          </FieldsContainer>
+        </ModalContentContainer>
+      </Modal.Body>
+      <Modal.Footer>
+        <ButtonContainer>
+          <Button bsStyle="link" onClick={_resetFavoriteField}>
+            Reset fields configuration
+          </Button>
```

Comment:
When I used this for the first time I was not sure to which state this would reset the configuration to. Maybe let's call it something like "Reset to default"?